src/domain/page-resolver.js

```diff
@@ -0,0 +1,81 @@
+import { PAGE_NAME } from './page-name.js';
+import { isBrowser } from '../ui/browser-helper.js';
+import DefaultPageTemplateComponent from '../components/page.js';
+
+const pageImporters = {
+  [PAGE_NAME.doc]: async () => (await import('../components/pages/doc.js')).default,
+  [PAGE_NAME.docs]: async () => (await import('../components/pages/docs.js')).default,
+  [PAGE_NAME.login]: async () => (await import('../components/pages/login.js')).default,
+  [PAGE_NAME.users]: async () => (await import('../components/pages/users.js')).default,
+  [PAGE_NAME.index]: async () => (await import('../components/pages/index.js')).default,
+  [PAGE_NAME.search]: async () => (await import('../components/pages/search.js')).default,
+  [PAGE_NAME.import]: async () => (await import('../components/pages/import.js')).default,
+  [PAGE_NAME.article]: async () => (await import('../components/pages/article.js')).default,
+  [PAGE_NAME.account]: async () => (await import('../components/pages/account.js')).default,
+  [PAGE_NAME.editDoc]: async () => (await import('../components/pages/edit-doc.js')).default,
+  [PAGE_NAME.settings]: async () => (await import('../components/pages/settings.js')).default,
+  [PAGE_NAME.register]: async () => (await import('../components/pages/register.js')).default,
+  [PAGE_NAME.resetPassword]: async () => (await import('../components/pages/reset-password.js')).default,
+  [PAGE_NAME.completeRegistration]: async () => (await import('../components/pages/complete-registration.js')).default,
+  [PAGE_NAME.completePasswordReset]: async () => (await import('../components/pages/complete-password-reset.js')).default
+};
+
+async function resolveAll(promiseCreators) {
+  // eslint-disable-next-line no-process-env
+  if (!isBrowser() && process.env.EDUCANDU_TEST === true.toString()) {
+    // Resolve sequentially in the test runner,
+    // because otherwise Jest bursts into flames:
+    // https://github.com/facebook/jest/issues/11434
+    const results = [];
+    for (const promiseCreator of promiseCreators) {
+      // eslint-disable-next-line no-await-in-loop
+      results.push(await promiseCreator());
+    }
+    return results;
+  }
+
+  // Otherwise we can resolve everything in parallel:
+  return Promise.all(promiseCreators.map(promiseCreator => promiseCreator()));
+}
+
+export default class PageResolver {
+  constructor(bundleConfig) {
```

Comment:
Because I think it is not to be decided by the person that configures the app, rather internal mechanics. The server caches (because it renders a lot of pages), the client not (there is only one page there, caching would make it even slower).